### PR TITLE
CMR-8506 Granule bulk update for ECHO10 OnlineResourceURL

### PR DESCRIPTION
**Pull request by jaybarra**

### CODE REVIEW

File: system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                            :granule gran})
+    (f)
+    (reset! test-atom {})))
+
+(use-fixtures :each (join-fixtures [(ingest/reset-fixture {"provguid1" "PROV1"})
+                                    ingest-granule-fixture]))
+
+(def no-hits? (complement string/includes?))
+
+(deftest online-resource-url-update-test--echo10--error-cases
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj:39]: nit: remove the extra `-`s from function name.

File: system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            status-response (ingest/granule-bulk-update-task-status task-id status-req-options)
+            {:keys [task-status status-message granule-statuses]} status-response]
+        (is (= "COMPLETE" task-status))
+        (is (= "Task completed with 1 FAILED out of 1 total granule update(s)." status-message))
+        (is (= [{:granule-ur "mime-type-update-gran-ur"
+                 :status "FAILED"
+                 :status-message "Update failed - please only specify URLs contained in the existing granule OnlineResources or OnlineAccessURLs [https://no-matching-url] were not found"}]
+               granule-statuses))))))
+
+(deftest online-resource-url-update-test--echo10--success-case
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj:80]: nit: remove the extra `-`s from function name.

File: system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      (is (= 200 status))
+      (is (some? task-id))
+      (let [status-req-options {:query-params {:show_granules "true"}}
+            status-response (ingest/granule-bulk-update-task-status task-id status-req-options)
+            {:keys [task-status status-message granule-statuses]} status-response]
+        (is (= "COMPLETE" task-status))
+        (is (= "Task completed with 1 FAILED out of 1 total granule update(s)." status-message))
+        (is (= [{:granule-ur "mime-type-update-gran-ur"
+                 :status "FAILED"
+                 :status-message "Update failed - please only specify URLs contained in the existing granule OnlineResources or OnlineAccessURLs [https://no-matching-url] were not found"}]
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj:77]: From this error message, it seems the update OnlineResourceURL can update either OnlineResources or OnlineAccessURLs. We only have test verifying update OnlineResources URL below. We need to add a test that verifies update OnlineAccessURLs.

**Review by ygliuvt — CHANGES_REQUESTED** [review]
It is unclear if Update OnlineResourceUrls will update granule's OnlineAccessURLs or not if there is a match. We need to either fix the misleading error message if OnlineAccessURLs is not supported or add a test if it is.

**Review by ygliuvt — APPROVED** [review]

File: system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj (hunk 3 above)
Comment (jaybarra) [reply system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj]: Fixed wording of the error message

**Review by isja17 — APPROVED** [review]

**Pull request merged**